src/main/java/io/stargate/sgv2/jsonapi/service/schema/collections/CollectionRerankDef.java

```diff
@@ -0,0 +1,414 @@
+ *
+ * <p>This class serves three main purposes in the collection lifecycle:
+ *
+ * <ol>
+ *   <li>During collection creation: Validates and transforms user-provided configuration
+ *       (deserialized from API as {@link CreateCollectionCommand.Options.RerankDesc}) into a
+ *       validated internal representation via {@link #fromApiDesc}.
+ *   <li>For persistence: After validation, instances of this class are serialized to JSON and
+ *       stored in the database as part of the collection comment.
+ *   <li>During query operations: Deserializes stored configuration in the comment via {@link
+ *       #fromCommentJson} to be used during reranking operations.
+ * </ol>
+ *
+ * <p>Note: This class is for internal use only and is not directly exposed through the DataAPI. The
+ * DataAPI exposes {@link CreateCollectionCommand.Options.RerankDesc} for user input, which is then
+ * validated and transformed into instances of this class before being persisted in the collection's
+ * metadata comment.
+ */
+public class CollectionRerankDef {
+  /**
+   * Singleton instance for disabled reranking configuration. It can be used for disabled reranking
+   * collections, existing pre-reranking collections, and missing collections.
+   */
+  public static final CollectionRerankDef DISABLED = new CollectionRerankDef(false, null);
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CollectionRerankDef.class);
+
+  private final boolean enabled;
+  private final RerankServiceDef rerankServiceDef;
+
+  /**
+   * Constructs a reranking configuration with the specified enabled state and service
+   * configuration.
+   *
+   * <p>This constructor is annotated with {@link JsonCreator} to enable Jackson deserialization.
+   * {@link JsonProperty} annotations on parameters allow Jackson to map JSON properties to
+   * constructor parameters during deserialization.
+   *
+   * <p>Validation behavior:
+   *
+   * <ul>
+   *   <li>If reranking is enabled (enabled = true), the service definition must not be null
+   *   <li>If reranking is disabled (enabled = false), the service definition is set to null
+   *       regardless of the input value to ensure consistency
+   * </ul>
+   *
+   * @param enabled Whether reranking is enabled for this collection
+   * @param rerankServiceDef The service configuration for reranking, must not be null if reranking
+   *     is enabled
+   * @throws NullPointerException if reranking is enabled and rerankingServiceConfig is null
+   */
+  @JsonCreator
+  public CollectionRerankDef(
+      @JsonProperty("enabled") boolean enabled,
+      @JsonProperty("service") RerankServiceDef rerankServiceDef) {
+    this.enabled = enabled;
+    this.rerankServiceDef =
+        enabled
+            ? Objects.requireNonNull(
+                rerankServiceDef,
+                "Rerank service configuration must not be null when reranking is enabled")
+            : null;
```

Comment:
I think this should throw if the rerankServiceDef is non-null , rather than silently drop. Sync with @tatu to make the same change in lexical config